### DISCUSSION TITLE
suggestions for funding sources and licensing

### DISCUSSION THREAD

**Discussion by programmerjake** [rui314/mold#861] [General]
I noticed that you wanted to relicense mold to a proprietary license due to a lack of funding, I think keeping mold open-source is highly important and changing to a proprietary license will stop a large portion of users on ethical grounds due to the license not being FOSS. Therefore I think it's worth spending a bit more time looking for funding that is compatible with keeping mold open-source. I have some suggestions for funding sources:
* [NLNet](https://nlnet.nl) -- they provide 10k-50k EUR (initially, up to 250k EUR later) grants funded by the EU for fully open-source/libre software/hardware as long as it provides benefits for europeans, imho mold probably counts because europeans benefit from having a faster linker. I have personal experience with NLNet, they have been funding [Libre-SOC](https://libre-soc.org/) (I am one of the founding members) for the past 3.5 years with around EUR 350k-400k from multiple grants.
* [FUTO](https://futo.org/) -- idk if mold qualifies, but it never hurts to try.
* [Rust Foundation Grants](https://foundation.rust-lang.org/grants/) -- mold probably qualifies because it provides a definite benefit to Rust.

**Comment by rui314** [comment]
Thank you for the links! As to government-backed funding,  if it comes with strings attached such as an obligation to keep it open-source, I don't want to take it. I'm developing this with a business interest in mind, and honestly €100k/yr isn't a very large sum. I couldn't hire an engineer after paying to myself, for example. I think we need a way to make commercial users who are actually benefited fro my work to pay for me.

**Reply by DemiMarie** [reply]
What about having support for open source platforms (Linux (including Android), \*BSD) be open source, while support for proprietary platforms (macOS, iOS, Windows) is under a proprietary license?  That is the [approach taken by SQLx](https://github.com/launchbadge/sqlx/discussions/909), and my understanding is that it has both worked out well for them financially and been very well-received by the community.  Obviously, it is your call, but I think this could be a best-of-both-worlds approach in many ways.  Distributions could continue to package mold, while the ones who need commercial licenses would be (to a good approximation) those already using proprietary OSs.
